Description: Declare possibly clobbered variables as 'volatile'.
 With '-O2' and '-Wextra' (wich are defined in uwsgiconfig.py) on
 GCC v4.4.5 20100902 (prerelease) some warnings (treated as errors) are
 generated in compiling ugreen.c (variable X might be clobbered by 'longjmp'
 or 'vfork').
 .
 This patch provide temporary fix of problem.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Forwarded: http://lists.unbit.it/pipermail/uwsgi/2010-September/000656.html
Last-Update: 2010-09-12

Index: uwsgi-0.9.6.1/ugreen.c
===================================================================
--- uwsgi-0.9.6.1.orig/ugreen.c	2010-09-12 17:27:27.000000000 +0300
+++ uwsgi-0.9.6.1/ugreen.c	2010-09-12 17:44:40.000000000 +0300
@@ -284,10 +284,10 @@
 
 void u_green_init(struct uwsgi_server *uwsgi) {
 
-	struct wsgi_request *wsgi_req = uwsgi->wsgi_requests ;
+	struct wsgi_request *volatile wsgi_req = uwsgi->wsgi_requests ;
 
-	int i;
-	size_t u_stack_size = UGREEN_DEFAULT_STACKSIZE ;
+	volatile int i;
+	volatile size_t u_stack_size = UGREEN_DEFAULT_STACKSIZE ;
 
 
 	PyMethodDef *uwsgi_function;
